packages/palette_generator/lib/palette_generator.dart

```diff
@@ -66,11 +66,10 @@ class PaletteGenerator with Diagnosticable {
   /// [PaletteGenerator.fromImage] static function. This constructor is mainly
   /// used for cases when you have your own source of color information and
   /// would like to use the target selection and scoring methods here.
-  ///
-  /// The [paletteColors] argument must not be null.
-  PaletteGenerator.fromColors(this.paletteColors, {this.targets})
-      : assert(paletteColors != null),
-        selectedSwatches = <PaletteTarget, PaletteColor>{} {
+  PaletteGenerator.fromColors(
+    this.paletteColors, {
+    this.targets = const <PaletteTarget>[],
+  }) : selectedSwatches = <PaletteTarget, PaletteColor>{} {
```

Comment:
Assert that `paletteColors` is non-empty here, and add that to the docs.

Reply:
Also remove the default value from this in `debugFillProperties` below (I can't comment on that line in GitHub):
```
    properties.add(IterableProperty<PaletteColor>(
        'paletteColors', paletteColors,
        defaultValue: <PaletteColor>[]));
```